docs/topics/compatibility-guide-15.md

```diff
@@ -0,0 +1,279 @@
+>  `-Xuse-old-backend` or `-language-version 1.4` can be used to temporarily revert to pre-1.5 behavior
+
+### Turn warnings into errors for calls with type parameters annotated by @OnlyInputTypes
+
+> **Issue**: [KT-45861](https://youtrack.jetbrains.com/issue/KT-45861)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.5 will outlaw calls like `contains`, `indexOf`, and `assertEquals` with senseless arguments to improve type safety
+>
+> **Deprecation cycle**:
+>
+> - 1.4.0: introduce warning for the problematic constructors
+> - \>= 1.5: raise this warning to an error,
+>  `-XXLanguage:-StrictOnlyInputTypesChecks` can be used to temporarily revert to pre-1.5 behavior
+
+### Use the correct order of arguments execution in calls with named vararg
+
+> **Issue**: [KT-17691](https://youtrack.jetbrains.com/issue/KT-17691)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Kotlin 1.5 will change the order of arguments execution in calls with named vararg
+>
+> **Deprecation cycle**:
+>
+> - < 1.5: introduce warning for the problematic constructors
+> - \>= 1.5: raise this warning to an error,
+>  `-XXLanguage:-UseCorrectExecutionOrderForVarargArguments` can be used to temporarily revert to pre-1.5 behavior
+
+### Use default value of the parameter in operator functional calls
+
+> **Issue**: [KT-42064](https://youtrack.jetbrains.com/issue/KT-42064)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Kotlin 1.5 will use default value of the parameter in operator calls
+>
+> **Deprecation cycle**:
+>
+> - < 1.5: old behavior (see details in the issue)
+> - \>= 1.5: behavior changed,
+>  `-XXLanguage:-JvmIrEnabledByDefault` can be used to temporarily revert to pre-1.5 behavior
+
+### Produce empty reversed progressions in for loops if regular progression is also empty 
+
+> **Issue**: [KT-42533](https://youtrack.jetbrains.com/issue/KT-42533)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Kotlin 1.5 will produce empty reversed progressions in for loops if regular progression is also empty
+>
+> **Deprecation cycle**:
+>
+> - < 1.5: old behavior (see details in the issue)
+> - \>= 1.5: behavior changed,
+>  `-XXLanguage:-JvmIrEnabledByDefault` can be used to temporarily revert to pre-1.5 behavior
+
+### Straighten Char-to-code and Char-to-digit conversions out
+
+> **Issue**: [KT-23451](https://youtrack.jetbrains.com/issue/KT-23451)
+>
+> **Component**: kotlin-stdlib
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Since Kotlin 1.5, conversions of Char to number types will be deprecated
+>
+> **Deprecation cycle**:
+>
+> - 1.5: deprecate `Char.toInt()/toShort()/toLong()/toByte()/toDouble()/toFloat()` and the reverse functions like `Int.toChar()`, and propose replacement 
```

Comment:
We've decided to leave `Int.toChar` not deprecated, so let's choose another example, e.g. `Long.toChar` or `Double.toChar`.